RELEASE.md

```diff
@@ -1,78 +1,234 @@
+### a. Make changes in codebase
 
-Open pull request to promote staging image:
-[add sha - version](https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-ingress-nginx/images.yaml#L1)
+- Make changes as per issue
 
-The sha is available in output from [cloud build](https://console.cloud.google.com/cloud-build/builds?project=k8s-staging-ingress-nginx)
+### b. Make changes to appropriate files in [images directory ](images)
 
-2. Change to images:
+- Make changes in /images 
+
+### c. Create PR
+
+- Open a PR to fire cloudbuild of a new image for the Ingress-Nginx-Controller
+
+  - In case of rare CVE fix or other reason to rebuild the nginx-base-image itself, look at the /images directory  [NGINX Base Image](https://github.com/kubernetes/ingress-nginx/tree/main/images/nginx)
+
+  - Example [NGINX_VERSION](images/nginx/rootfs/build.sh#L21),
+
+  - [SHA256](images/nginx/rootfs/build.sh#L124) 
+
+  - If you are updating any component in [build.sh](images/nginx/rootfs/build.sh) please also update the SHA256 checksum of that component as well, the cloud build will fail with an exit 10 if not.  
+
+### d. Merge
+
+- Merging success should fire cloudbuild
+
+### e. Make sure cloudbuild is a success
+
+- Wait for [cloud build](https://console.cloud.google.com/cloud-build/builds?project=k8s-staging-ingress-nginx)
+
+- Proceed only after cloud-build is successful in building a new Ingress-Nginx-Controller image.
+
+
+## 2. If applicable, BUILD other images
+
+- If applicable, then build a new image of any other related component, ONLY IF APPLICABLE TO THE RELEASE
+
+### a. If applicable then make changes in relevant codebase
+
+- Change code as per issue
+
+### b. Make changes to appropriate files in [images directory ](images)
+
+- Sometimes, you may also be needing to rebuild, images for one or multiple other related components of the Ingress-Nginx-Controller ecosystem. Make changes to the required files in the /images directory, if/as applicable, in the context of the release you are attempting.  :
    
-* [e2e](https://github.com/kubernetes/ingress-nginx/tree/main/images/test-runner)
+  - [e2e](https://github.com/kubernetes/ingress-nginx/tree/main/test/e2e-image)
+
+    - Update references to e2e-test-runner image [If applicable] :
+
+      - [e2e-image](https://github.com/kubernetes/ingress-nginx/blob/main/test/e2e-image/Dockerfile#L1)
+      - [run-in-docker.sh](https://github.com/kubernetes/ingress-nginx/blob/main/build/run-in-docker.sh#L37)
+
+  - [test-runner](https://github.com/kubernetes/ingress-nginx/tree/main/images/test-runner)
+
+  - [echo](https://github.com/kubernetes/ingress-nginx/tree/main/images/echo)
+
+  - [cfssl](https://github.com/kubernetes/ingress-nginx/tree/main/images/cfssl)
+
+  - [fastcgi-helloserver](https://github.com/kubernetes/ingress-nginx/tree/main/images/fastcgi-helloserver)
+
+  - [httpbin](https://github.com/kubernetes/ingress-nginx/tree/main/images/httpbin)
+
+  - kube-webhook-certgen
```

Comment:
Please link to the image like the others.

Reply:
I have to make te effort to figure out where we forked and adapted the certgen code. Since it is not recurring, we can do this in a follow up PR